lib/ansible/modules/source_control/git.py

```diff
@@ -354,17 +354,23 @@ def get_submodule_update_params(module, git_path, cwd):
     return params
 
 
-def write_ssh_wrapper():
+def write_ssh_wrapper(module):
     module_dir = get_module_path()
     try:
         # make sure we have full permission to the module_dir, which
         # may not be the case if we're sudo'ing to a non-root user
         if os.access(module_dir, os.W_OK | os.R_OK | os.X_OK):
-            fd, wrapper_path = tempfile.mkstemp(prefix=module_dir + '/')
+            if module._tmpdir is not None:
+                fd, wrapper_path = tempfile.mkstemp(prefix=module._tmpdir + '/')
+            else:
+                fd, wrapper_path = tempfile.mkstemp(prefix=module_dir + '/')
```

Comment:
this is basically noop, unless you are under pipelining, in which you get None, tmpdir will still be populated though

Reply:
Yeah I do want to believe you, and it's precisely for this kind of reason why I have purpose firstly a simple twick fix with an option at none by default, because I don't have time actually for read undocumented core / private module for fixe a problem that has been around for at least a year, I just start with ansible and git is obviously one of the most usefull module for deployement, so if you don't purpose other thing, I can't use more time actually on this problem, so I just go to use this probably bad hack as long as it allows me to run this module on servers whose exec flags will never be allowed even temporarily on / tmp. And sorry for the disturbance...